src/test/java/mate/academy/springbootintro/controller/ShoppingCartControllerTest.java

```diff
@@ -0,0 +1,258 @@
+import org.junit.jupiter.api.Assertions;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithUserDetails;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ShoppingCartControllerTest {
+
+    protected static MockMvc mockMvc;
+    private static final String SPRING_DATASOURCE_URL =
+            "spring.datasource.url";
+    private static final String SPRING_DATASOURCE_USERNAME =
+            "spring.datasource.username";
+    private static final String SPRING_DATASOURCE_PASSWORD =
+            "spring.datasource.password";
+    private static final String SPRING_DATASOURCE_DRIVER_CLASS_NAME =
+            "spring.datasource.driver-class-name";
+    private static final String DEFAULT_USER_EMAIL = "user@example.com";
+    private static final int QUANTITY = 25;
+    private static final Long USER_ID = 2L;
+    private static final Long BOOK_ID = 3L;
+    private static final Long CART_ITEM_ID = 1L;
+    private static final Long CART_ITEM_DTO_ID = 1L;
+    private static final Long SHOPPING_CART_ID = 2L;
+    private static final Long SHOPPING_CART_DTO_ID = 2L;
+    private static final String EXPECTED_BOOK_TITLE = "Game of thrones";
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @Autowired
+    private CartItemRepository cartItemRepository;
+
+    @DynamicPropertySource
+    static void configureProperties(DynamicPropertyRegistry registry) {
+        CustomMySqlContainer mysqlContainer = CustomMySqlContainer.getInstance();
+        registry.add(SPRING_DATASOURCE_URL, mysqlContainer::getJdbcUrl);
+        registry.add(SPRING_DATASOURCE_USERNAME, mysqlContainer::getUsername);
+        registry.add(SPRING_DATASOURCE_PASSWORD, mysqlContainer::getPassword);
+        registry.add(SPRING_DATASOURCE_DRIVER_CLASS_NAME, mysqlContainer::getDriverClassName);
+    }
+
+    @BeforeAll
+    static void beforeAll() {
+        CustomMySqlContainer.getInstance().start();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        CustomMySqlContainer.getInstance().stop();
+    }

```

Comment:
Also redundant, please refer to Mate materials to check what should be done in the `afterAll` and `beforeAll` methods!